### PR TITLE
Remove TensorVariable Init

### PR DESCRIPTION
**Pull request by skaae**
Removes TensorVariable special case as discussed here: https://github.com/Lasagne/Lasagne/issues/462

* Removes TensorVariable Init.
* Removes related Tests

### CODE REVIEW

**Comment by skaae** [comment]
I noticed that `hid_init` does not have a reference to `lasagne.layers.Layer` .
`:class:lasagne.layers.Layer` does not fit the linewidth but :class:Layer` does.

Should I change add `:class:Layer` and maybe also change the rest of the recurrent docs to use
`:class:Layer` because its more consistent with the rest of the docs?

**Comment by f0k** [comment]
> Removes TensorVariable special case

Looks good!

> Should I change add :class:Layer and maybe also change the rest of the recurrent docs to use
:class:Layer because its more consistent with the rest of the docs?

Yes, would be good to change that! :class:Layer is enough because we tell Sphinx that we're in the lasagne.layers namespace. The docstring change should be in a separate commit this time, i.e., not squashed into the first one. Let me know when you're done so we can merge! (github doesn't send out a notification for additional commits to a PR, only for additional comments.)

**Comment by skaae** [comment]
I added references to the layer class in the doc string and squashed the commits.

File: lasagne/layers/recurrent.py [hunk 1]
```diff
@@ -748,22 +730,16 @@ class LSTMLayer(MergeLayer):
...
         None is provided, no nonlinearity will be applied.
-    cell_init : callable, np.ndarray, theano.shared, TensorVariable or Layer
-        Initializer for initial cell state (:math:`c_0`).  If a
-        TensorVariable (Theano expression) is supplied, it will not be learned
-        regardless of the value of `learn_init`.
-    hid_init : callable, np.ndarray, theano.shared, TensorVariable or Layer
-        Initializer for initial hidden state (:math:`h_0`).  If a
-        TensorVariable (Theano expression) is supplied, it will not be learned
-        regardless of the value of `learn_init`.
+    cell_init : callable, np.ndarray, theano.shared or Layer
```
Comment (f0k) [inline lasagne/layers/recurrent.py]: you missed that one!

**Comment by f0k** [comment]
One `Layer` reference missed, other than that, it's good to merge. Can you please fix that (and squash)?

**Comment by skaae** [comment]
Sorry about that. I fixed it.

**Comment by f0k** [comment]
Thanks, merging!

**Pull request merged**